test/radium-test.js

```diff
@@ -6,13 +6,6 @@ const Radium = require('..');
 const render = require('./utils').render;
 
 describe('Radium blackbox SSR tests', () => {
-  it('sets up initial state', () => {
```

Comment:
didn't really feel like  much of a test, we're not even using react here despite this file appearing to be an SSR/integration test.  We have this unit test in the enhancer karma test. I can put this back as something a little more meaningful if you like.

I have to rework it a little though, I can't just `new` the resulting component anymore (see changes to other tests that previously did `new Enhanced()`)

Reply:
If this is true, then have a semver major change I believe.

Reply:
Also note that classes in `test/` are **real** classes via Node, whereas anything in Karma (`src/__tests__`) are fake transpiled Babel classes (and the impetus for creating the `test/` files and harness in the first place was the differences found only in real classes)...

Don't know that this necessarily means we need to keep this particular test, but if it tests / asserts on real classes, we should support it...

Reply:
Hey @ryan-roemer,

With the switch to use the new context API, for class components wrapped in `Radium()`, the 2 contexts are now injected via a hoc which is a function component (which in turn uses `useContext` to get the contexts).

What this means, is that `new Xxxx` does not give you direct access to your extended class (notice in the karma tests I had passed `ref` in instead, alternatively you could reach in to the child nodes in the render output). So to actually check the state, you need to render the wrapped component with React.

While there probably are changes that warrant a semver change (although, `0.y.z` is free game for breaking changes as far as semver is concerned without bumping the major version number), I'm not sure this particular symptom is one that warrants it. The public API surface in this respect hasn't changed, as doing `new MyComponent()` isn't how you get your component rendered onto a page, and it's my understanding  that Radium promises that the result of `Radium(MyComponent)` is a component  that  can be rendered by React (that renders your component as it would but with the styling functionality Radium provides). It was just being `new`d in tests to provide  a convenient way to check the instance properties as far as I can tell, that's why I felt it was okay to change this pattern in other places to grab the component instance by passing a `ref` instead.

Reply:
Also, at least for me locally, these tests run on the transpiled version in `lib`:

```
yarn test-node-dev
yarn run v1.12.1
$ mocha --watch "test/**/*-test.js"

Error: Cannot find module '../lib/enhancer'
```

Reply:
Yep, our CONTRIBUTING doc already notes this: https://github.com/FormidableLabs/radium/blob/master/CONTRIBUTING.md#backend you need a babel watch running in a different terminal / process to use mocha --watch

The background here is we need to make sure we're running in real Node with real classes, etc. (as opposed to babel-node which is mostly the same as transpiling the test code...)

Reply:
Hey Ryan, sorry, I just wiped my lib folder to demonstrate it was looking for lib, which contains transpiled code not native classes.

Reply:
Yeah, lib is transpiled as the docs say. These tests are not and use native classes. We don’t separately ship native code for node in lib.

Reply:
I was highlighting because you had previously said

> Also note that classes in `test/` are **real** classes via Node, whereas anything in Karma (`src/__tests__`) are fake transpiled Babel classes (and the impetus for creating the test/ files and harness in the first place was the differences found only in real classes)...

Which doesn't seem to be the case (unless I misunderstood your comment).

Either way -- I think it is besides the point, since, real class or not, the enhanced component class is wrapped in a function hoc now.

Reply:
Nothing in “test/“ is transpiled is what that says. Everything in lib is. Does that help resolve the confusion?

Reply:
The entire point of the node tests is mainly to not transpile 😉

Reply:
The entire point of the node tests is mainly to not transpile 😉

Reply:
The entire point of the node tests is mainly to not transpile 😉

Reply:
Gotcha. So, the point is to use transpiled Radium, but in node with real classes from your own codebase? (which makes total sense to test)

However, this test case though is def irrelevant for validating that, as the test case doesn't create a class, it just passes `React.Component` into `Radium`.

Reply:
Yep on both points! (Sorry that the docs clearly aren’t explaining the distinction clearly enough as you’re in the weeds of the project now and it’s still confusing...)

Reply:
No worries! It's probably my fault for not spending enough time familiarizing myself with the test setup and the reasons why things are done a certain way.

Wonder if it's beneficial to try set some similar tests up for the browser now too with native classes + transpiled code 🤔

Reply:
> Wonder if it's beneficial to try set some similar tests up for the browser now too with native classes + transpiled code

Yep! Also on potentially same topic I do wonder if we have lingering issues like https://github.com/FormidableLabs/radium/issues/1017 that may be due to browser use of `<script type="module">` (and in that case because we have babel generated helpers above the first `import` statement)...

Reply:
Resolving this, as either way we determined that this test case is irrelevant for the purpose of these tests.